File: dsl-style/src/test/java/example/dsl_style/InsertTest.java
package example.dsl_style;

import example.dsl_style.domain.Age;
import example.dsl_style.domain.Salary;
import example.dsl_style.entity.Employee;
import example.dsl_style.entity.JobType;
import example.dsl_style.repository.EmployeeRepository;
import java.sql.Timestamp;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith(TestEnvironment.class)
public class InsertTest {

  private final EmployeeRepository repository;

  InsertTest(DbConfig config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testInsert() {
    var employee = new Employee();
    employee.setName("test");
    employee.setAge(new Age(50));
    employee.setSalary(new Salary(300));
    employee.setJobType(JobType.PRESIDENT);
    repository.insert(employee);
  }

  @Test
  public void testInsertByNativeSql() {
    var employee = new Employee();
    employee.setId(100);
    employee.setName("test");
    employee.setAge(new Age(50));
    employee.setSalary(new Salary(300));
    employee.setJobType(JobType.PRESIDENT);
    employee.setInsertTimestamp(new Timestamp(System.currentTimeMillis()));
    employee.setVersion(1);
    repository.insertByNativeSql(employee);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl_style_kotlin/InsertTest.kt
package example.dsl_style_kotlin

import example.dsl_style_kotlin.domain.Age
import example.dsl_style_kotlin.domain.Salary
import example.dsl_style_kotlin.entity.Employee
import example.dsl_style_kotlin.entity.JobType
import example.dsl_style_kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config
import java.sql.Timestamp

@ExtendWith(TestEnvironment::class)
class InsertTest internal constructor(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testInsert() {
        val employee = Employee()
        employee.name = "test"
        employee.age = Age(50)
        employee.salary = Salary(300)
        employee.jobType = JobType.PRESIDENT
        repository.insert(employee)
    }

    @Test
    fun testInsertByNativeSql() {
        val employee = Employee()
        employee.id = 100
        employee.name = "test"
        employee.age = Age(50)
        employee.salary = Salary(300)
        employee.jobType = JobType.PRESIDENT
        employee.insertTimestamp = Timestamp(System.currentTimeMillis())
        employee.version = 1
        repository.insertByNativeSql(employee)
    }
}
